engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoDateTime.java

```diff
@@ -72,11 +71,18 @@ public static EnsoDateTime now() {
   @Builtin.WrapException(from = DateTimeParseException.class)
   @CompilerDirectives.TruffleBoundary
   public static EnsoDateTime parse(String text) {
-    TemporalAccessor time = TIME_FORMAT.parseBest(text, ZonedDateTime::from, LocalDateTime::from);
-    if (time instanceof ZonedDateTime) {
-      return new EnsoDateTime((ZonedDateTime) time);
-    } else if (time instanceof LocalDateTime) {
-      return new EnsoDateTime(((LocalDateTime) time).atZone(ZoneId.systemDefault()));
+    String iso = text;
+    if (text != null && text.length() > 10 && text.charAt(10) == ' ') {
+      var builder = new StringBuilder(iso);
+      builder.replace(10, 11, "T");
+      iso = builder.toString();
```

Comment:
I hope my try at the shared Java project between `runtime` and `std-bits` will allow us to avoid this code duplication.

Reply:
True, they will be. But I don't think this is an issue at all - firstly we should only put the very core stuff that needs to be shared, so it should have minimal footprint.

Secondly, it will still have _not greater_ footprint than duplicated code.

And ensuring we do not duplicate code will increase maintability - we will have a single source of truth for these implementations - easier to update (as otherwise it is easy to forget to update one of the copies when doing changes).